chameleon/src/test/java/org/hibernate/omm/query/OrderByTests.java

```diff
@@ -0,0 +1,71 @@
+package org.hibernate.omm.query;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import org.hibernate.omm.AbstractMongodbIntegrationTests;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author Nathan Xu
+ */
+class OrderByTests extends AbstractMongodbIntegrationTests {
+
+    @Test
+    void testOrderBy() {
+        var book1 = new Book();
+        book1.id = 1L;
+        book1.title = "War and Peace";
+        book1.author = "Leo Tolstoy";
+        book1.publishYear = 1869;
+
+        var book2 = new Book();
+        book2.id = 2L;
+        book2.title = "Crime and Punishment";
+        book2.author = "Leo Tolstoy";
+        book2.publishYear = 1869;
+
+        getSessionFactory().inTransaction(session -> {
+            session.persist(book1);
+            session.persist(book2);
+        });
+
+        getSessionFactory().inSession(session -> {
+            var query = session.createQuery("from Book b order by b.title asc", Book.class);
+            var books = query.getResultList();
+            assertThat(books).usingRecursiveComparison().isEqualTo(List.of(book2, book1));
+
+            query = session.createQuery("from Book b order by b.title desc", Book.class);
+            books = query.getResultList();
+            assertThat(books).usingRecursiveComparison().isEqualTo(List.of(book1, book2));
+
+            query = session.createQuery("from Book b order by b.publishYear, b.title desc", Book.class);
+            books = query.getResultList();
+            assertThat(books).usingRecursiveComparison().isEqualTo(List.of(book1, book2));
+        });
+    }
+
+    @Override
+    public List<Class<?>> getAnnotatedClasses() {
+        return List.of(Book.class);
+    }
+
+    @Entity(name = "Book")
+    @Table(name = "books")
+    static class Book {
+
+        @Id
+        Long id;
+
+        String title;
+
+        String author;
+
+        int publishYear;
+
+    }
+}
```

Comment:
In recent code changes, the following testing paradigm was introduced in lieu of the above inheritance pattern:
```
@MongoIntegrationTest
class SomeTests {

    @SessionFactoryInjected
    SessionFactory sessionFactory; // static or not is ok
    
    // if needed
    @MongoDatabaseInjected
    MongoDatabase mongoDatabase; // static or not is ok

    @Test
    void test() {
        ... ...
    }
    
    @Entity(name = "Book")
    @Table(name = "books")
    static class Book {
        @Id
        Long id;

        Book() {}
        Book(final Long id) {
            this.id = id;
        }
    }
}
```

The benefits are many, including:

- no need to extend, so we maintain the freedom to extend other parent class (if needed)
- no need to specify inner entity classes which will be discovered automatically
- given entity class could be found before testing instance is launched, we don't need to create Hibernate's SessionFactory and Mongo container for each test method but create once during class loading phase, thus speeding up test perf to maximally

The idea was inspired by Steve's similar mechanism but overcame some of its limits. It needs polishment but it is easy to do